src/hdt.rs

```diff
@@ -81,6 +96,214 @@ impl Hdt {
+        headers.insert(Triple::new(
+            dict_id.clone(),
+            HDT_DICT_SIZE_STRINGS.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(
+                ByteSize(self.dict.size_in_bytes() as u64).to_string(),
+            )),
+        ));
+        // header.insert(rootNode, HDTVocabulary::DICTIONARY_BLOCK_SIZE, this->blocksize);
+        headers.insert(Triple::new(
+            dict_id,
+            HDT_DICT_BLOCK_SIZE.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(opts.block_size.to_string())),
+        ));
+
+        // TRIPLES
+        // header.insert(rootNode, HDTVocabulary::TRIPLES_TYPE, getType());
+        headers.insert(Triple::new(
+            triples_id.clone(),
+            DC_TERMS_FORMAT.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(HDT_TYPE_BITMAP.to_owned())),
+        ));
+        // header.insert(rootNode, HDTVocabulary::TRIPLES_NUM_TRIPLES, getNumberOfElements() );
+        headers.insert(Triple::new(
+            triples_id.clone(),
+            HDT_NUM_TRIPLES.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(num_triples.to_string())),
+        ));
+        // header.insert(rootNode, HDTVocabulary::TRIPLES_ORDER, getOrderStr(order) );
+        headers.insert(Triple::new(
+            triples_id,
+            HDT_TRIPLES_ORDER.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(opts.order)),
+        ));
+
+        // // Sizes
+        let meta = std::fs::File::open(f).unwrap().metadata().unwrap();
+        // header->insert(statisticsNode, HDTVocabulary::ORIGINAL_SIZE, origSize);
+        headers.insert(Triple::new(
+            stats_id.clone(),
+            HDT_ORIGINAL_SIZE.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(meta.len().to_string())),
+        ));
+        // header->insert(statisticsNode, HDTVocabulary::HDT_SIZE, getDictionary()->size() + getTriples()->size());
+        headers.insert(Triple::new(
+            stats_id,
+            HDT_SIZE.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(
+                ByteSize(self.size_in_bytes() as u64).to_string(),
+            )),
+        ));
+
+        // // Current time
+        // struct tm* today = localtime(&now);
+        // strftime(date, 40, "%Y-%m-%dT%H:%M:%S%z", today);
+        // header->insert(publicationInfoNode, HDTVocabulary::DUBLIN_CORE_ISSUED, date);
+        let now = chrono::Utc::now(); // Get current local datetime
+        let datetime_str = now.format("%Y-%m-%dT%H:%M:%S%z").to_string(); // Format as string
+        headers.insert(Triple::new(
+            pub_id,
+            DC_TERMS_ISSUED.to_owned(),
+            containers::rdf::Term::Literal(containers::rdf::Literal::new(datetime_str)),
+        ));
+
+        // TODO fix header length
```

Comment:
@KonradHoeffner pretty sure this TODO for the length is going to break the write and subsequent reads. I didn't check, but pretty sure the ControlInfo for the dictionary and triples are not getting added during the write yet